CMakeLists.txt

```diff
@@ -99,83 +99,134 @@ if(FCL_COVERALLS)
 endif()
 
 find_package(PkgConfig QUIET)
+set(PKG_CONFIG_USE_CMAKE_PREFIX_PATH ON)
 
-# Find Eigen3
-find_package(Eigen3 3.0.5 QUIET)
-if(EIGEN3_FOUND)
+#===============================================================================
+# Find required dependency Eigen3 (>= 3.0.5)
+#
+# If Eigen3 is not found, manually set the cache variable EIGEN3_INCLUDE_DIR
+#===============================================================================
+find_package(Eigen3 3.0.5 QUIET CONFIG)
+
+# If Eigen3Config.cmake is not found, use the FindEigen3.cmake module
+if(NOT Eigen3_FOUND)
+  find_package(Eigen3 3.0.5 QUIET MODULE)
+  set(Eigen3_FOUND ON)
+endif()
```

Comment:
Because CMake uses `MODULE` mode first and falls back to `CONFIG` mode, which unfortunately is the opposite that we would want.